frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -280,15 +285,15 @@ pub mod pallet {
             let force_new_era = Self::force_era().eq(&Forcing::ForceNew);
             let blocks_per_era = T::BlockPerEra::get();
             let previous_era = Self::current_era();
+            let next_era_starting_block = Self::next_era_starting_block();
 
             // Value is compared to 1 since genesis block is ignored
-            if now % blocks_per_era == BlockNumberFor::<T>::from(1u32)
-                || force_new_era
-                || previous_era.is_zero()
-            {
+            if now == next_era_starting_block || force_new_era || previous_era.is_zero() {
                 let next_era = previous_era + 1;
                 CurrentEra::<T>::put(next_era);
 
+                NextEraStartingBlock::<T>::put(now + blocks_per_era);
```

Comment:
One of the parts of this task is to benchmark the on_initialize method (check the github issue ticket).

* https://docs.substrate.io/v3/runtime/benchmarking/
* check the benchmarking.rs file in pallet-dapps-staking module